### ISSUE TITLE
Unit testing code with utilizes ConnectionMultiplexer

### ISSUE THREAD

**Issue by RPM1984** [StackExchange/StackExchange.Redis#230]
Hi,

So i have code like this:

```
public class SomeCacheLayer
{
	public ConnectionMultiplexer _connectionMultiplexer;

	public SomeCacheLayer(ConnectionMultiplexer connectionMultiplexer)
	{
		_connectionMultiplexer = connectionMultiplexer;
	}
	
	public void SomeMethodIWantToTest()
	{
		SomethingImportant();
		_connectionMultiplexer.GetDatabase().KeyExpire(TimeSpan.FromMinutes(10));
	}
```

The problem is, this class isn't testable, due to no interface existing for `ConnectionMultiplexer`.

I solved this by creating my own wrapper class.

Any reason why there is no `IConnectionMultiplexer` class?

Just to be clear, i'm not trying to test `ConnectionMultiplexer`, i'm trying to test my code which utilizes `ConnectionMultiplexer`.

Example test:

```
[Fact]
public void SomeTest()
{
    // Arrange.
    var connectionMultiplexer = A.Fake<IConnectionMultiplexer>();
    var cacheLayer = new SomeCacheLayer(connectionMultiplexer);

    // Act.
    cacheLayer.SomeMethodIWantToTest();

   // Assert.
   A.CallTo(() => connectionMultiplexer.GetDatabase().KeyExpire(A<TimeSpan>._)
      .MustHaveHappened();   
}
```

**Comment by nmehlei** [comment]
Duplicate of #176 , though still a very useful suggestion

**Comment by NickCraver** [comment]
Closing this out - addressed via #231's addition of `IConnectionMultiplexer`.
